src/main/java/seedu/address/logic/commands/view/ViewCompanyCommand.java

```diff
@@ -0,0 +1,80 @@
+package seedu.address.logic.commands.view;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_ITEM_DISPLAYED_INDEX;
+import static seedu.address.commons.core.Messages.MESSAGE_VIEW_SUCCESS;
+import static seedu.address.model.util.ItemUtil.COMPANY_ALIAS;
+import static seedu.address.model.util.ItemUtil.COMPANY_NAME;
+
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.company.CompanyItem;
+import seedu.address.ui.tabs.TabName;
+
+public class ViewCompanyCommand extends ViewCommand {
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + " " + COMPANY_ALIAS + ": Views a " + COMPANY_NAME + " in "
+            + "InternHunter.\nParameters: "
+            + "INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " "
+            + COMPANY_ALIAS
+            + " 3";
+
+    private final String messageViewSuccess;
+    private final String messageAlreadyViewing;
+    private final Index targetIndex;
+
+    /** todo javadocs (shawn) */
+    public ViewCompanyCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+        this.messageViewSuccess = String.format(MESSAGE_VIEW_SUCCESS, COMPANY_NAME, targetIndex);
+        this.messageAlreadyViewing = String.format(MESSAGE_ALREADY_VIEWING, COMPANY_NAME, targetIndex);
+    }
+
+    /**
+     * Executes the view company command and returns the result message.
+     * This command also auto-switches the user to the company tab.
+     *
+     * @param model {@code Model} which the command should operate on.
+     * @return feedback message of the operation result for display
+     * @throws CommandException If an error occurs during command execution.
+     */
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<CompanyItem> lastShownList = model.getCompanyList().getFilteredItemList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(String.format(MESSAGE_INVALID_ITEM_DISPLAYED_INDEX, COMPANY_NAME));
+        }
+
+        String resultMessage = messageViewSuccess;
+        boolean shouldSwitchTab = false;
+        boolean shouldSwitchDisplay = true;
```

Comment:
I think that it might be better if we initialize the variables in the if-else block, so that it is clearer which block will have what booleans. And also this will stick to the notion of defining the variables in the least possible scope!

Reply:
Hi Keane! Unfortunately that would not be possible. Variables can't be declared in an if-else block and then used outside of it due to variable scope. Even doing something as simple as:
```
if (true) {
    boolean shouldSwitchTab = false;
} else {
    boolean shouldSwitchTab = false;
}
shouldSwitchTab = true; // line X
```
Will lead to an error on line X whereby it doesn't detect that shouldSwitchTab has been initialized.
(More on it here: [https://stackoverflow.com/questions/17444020/why-cant-variables-be-declared-in-an-if-statement](https://stackoverflow.com/questions/17444020/why-cant-variables-be-declared-in-an-if-statement#:~:text=Because%20when%20b%20goes%20out,in%20the%20next%20if%20statement.&text=So%20here%20scope%20of%20b,it%20is%20giving%20compilation%20error.))

As such, this already is the smallest possible scope!

Reply:
To justify having these boolean vars initialized and declared where they are, we can take it that by default, we assume the user in on the appropriate tab (`shouldSwitchTab = false`) and not viewing the current item (`shouldSwitchDisplay = true`) which is why they would want to execute the view command in the first place. However, they may happen to be on the wrong tab (first if statement checks for that), or they may happen to be on the right tab and already viewing the item (the following else-if statement checks for that). That is how I would intend the code to be read!

Reply:
Oh what I meant was initializing the value in the if-else block and not declaring it there. So an example of this is:

![image](https://user-images.githubusercontent.com/65328119/95716054-62af8180-0c9d-11eb-88b7-a73726580bd4.png)

Reply:
Ah my bad! But then will we be violating this coding standard rule?
![image](https://user-images.githubusercontent.com/65944172/95719932-23842f00-0ca3-11eb-9dad-ed9f56eff405.png)

Reply:
Don't think so, since he did mention that this is just a guide and I remembered there was an example which he said would be ok?

Reply:
Oh if that's the case, sure I don't mind changing it! But then again I'm not sure if our evaluators have the same understanding though? For example when we eventually evaluate our peers' work, I would probably follow the coding standard to cross-check my evaluation and if I spotted this violation in their work, I would probably just mark it as a violation since it is stated in black and white and not have to think too much. If you have the source, could you point me to where Prof Damith mentioned that we could do otherwise? Would be much appreciated!